File: backend/src/main/java/com/yagubogu/auth/dto/LoginResponse.java
package com.yagubogu.auth.dto;

import com.yagubogu.member.domain.Member;

public record LoginResponse(
        String accessToken,
        String refreshToken,
        boolean isNew,
        MemberResponse user
) {
    public record MemberResponse(
            long id,
            String nickname,
            String profileImageUrl
    ) {
        public static MemberResponse from(final Member member) {
            return new MemberResponse(member.getId(), member.getNickname(), member.getImage());
        }
    }
}

File: android/app/src/main/java/com/yagubogu/data/dto/response/auth/LoginResponse.kt
package com.yagubogu.data.dto.response.auth

import com.yagubogu.data.dto.response.MemberDto
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class LoginResponse(
    @SerialName("accessToken")
    val accessToken: String, // access 토큰
    @SerialName("refreshToken")
    val refreshToken: String, // refresh 토큰
    @SerialName("isNew")
    val isNew: Boolean, // 회원가입 여부 (true: 회원가입, false: 로그인)
    @SerialName("member")
    val member: MemberDto, // 사용자 정보
)
